splunk-otel-android/src/main/java/com/splunk/rum/RumInitializer.java

```diff
@@ -286,45 +291,70 @@ private Resource createSplunkResource() {
     private void installAnrDetector(OpenTelemetryRumBuilder otelRumBuilder, Looper mainLooper) {
         otelRumBuilder.addInstrumentation(
                 instrumentedApplication -> {
-                    AnrDetector.builder()
-                            .addAttributesExtractor(constant(COMPONENT_KEY, COMPONENT_ERROR))
-                            .setMainLooper(mainLooper)
-                            .build()
-                            .installOn(instrumentedApplication);
+                    ErrorIdentifierExtractor extractor = new ErrorIdentifierExtractor(application);
+                    ErrorIdentifierInfo errorIdentifierInfo = extractor.extractInfo();
+                    String applicationId = errorIdentifierInfo.getApplicationId();
+                    String versionCode = errorIdentifierInfo.getVersionCode();
+                    String uuid = errorIdentifierInfo.getCustomUUID();
 
-                    initializationEvents.emit("anrMonitorInitialized");
-                });
-    }
+                    AnrDetectorBuilder builder = AnrDetector.builder();
+                    builder.addAttributesExtractor(constant(COMPONENT_KEY, COMPONENT_ERROR));
 
-    private void installSlowRenderingDetector(OpenTelemetryRumBuilder otelRumBuilder) {
-        otelRumBuilder.addInstrumentation(
-                instrumentedApplication -> {
-                    SlowRenderingDetector.builder()
-                            .setSlowRenderingDetectionPollInterval(
-                                    builder.slowRenderingDetectionPollInterval)
-                            .build()
-                            .installOn(instrumentedApplication);
-                    initializationEvents.emit("slowRenderingDetectorInitialized");
+                    if (applicationId != null)
+                        builder.addAttributesExtractor(constant(APPLICATION_ID_KEY, applicationId));
+                    if (versionCode != null)
+                        builder.addAttributesExtractor(constant(APP_VERSION_CODE_KEY, versionCode));
+                    if (uuid != null)
+                        builder.addAttributesExtractor(constant(SPLUNK_OLLY_UUID_KEY, uuid));
```

Comment:
Hmmmm, we should be requiring curly braces around ifs. Weird that the formatter check didn't get this.

Reply:
Added silly follow-up issue so this doesn't get blocked: https://github.com/signalfx/splunk-otel-android/issues/1015